File: scripts/algorithms/I/Island Perimeter/Island Perimeter.java
// Runtime: 10 ms (Top 73.55%) | Memory: 62.3 MB (Top 52.61%)
class Solution {
    public int islandPerimeter(int[][] grid) {
        if(grid == null || grid.length == 0) return 0;

        int row=grid.length,col=grid[0].length;
        int perimeter=0;

        for(int i=0;i<row;i++){
            for(int j=0;j<col;j++){

                if(grid[i][j]==1){
                    perimeter+=4;

                    if(i>0 && grid[i-1][j]==1){
                    perimeter-=2;
                   }

                    if(j>0 && grid[i][j-1]==1){
                    perimeter-=2;
                   }
                }

            }
        }
        return perimeter;
    }
}

File: scripts/algorithms/I/Island Perimeter/Island Perimeter.cpp
// Runtime: 75 ms (Top 80.5%) | Memory: 96.50 MB (Top 55.46%)

class Solution {
public:
    int islandPerimeter(vector<vector<int>>& grid) {
        int ans = 0 , row = grid.size(), col = grid[0].size();    
        for(int i = 0 ; i < row ; i++)
        {
           for(int j = 0 ; j < col ; j++)
           {
               if( grid[i][j])
               {
                  ans+=4;
                  if(i < row - 1 && grid[i+1][j])ans--; 
                  if(j < col -1 && grid[i][j+1])ans--; 
                  if(j  && grid[i][j-1])ans--; 
                  if(i  && grid[i-1][j])ans--; 
               }
           } 
        }
        return ans;
    }
};
